assert/assertions.go

```diff
@@ -502,7 +502,13 @@ func Same(t TestingT, expected, actual interface{}, msgAndArgs ...interface{}) b
 		h.Helper()
 	}
 
-	if !samePointers(expected, actual) {
+	same, ok := samePointers(expected, actual)
+	if !ok {
+		return Fail(t, "Both arguments must be pointers", msgAndArgs...)
+	}
+
+	//both are pointers but pointing to different addresses
+	if !same {
```

Comment:
This comment isn't true in that location, it also doesn't mention the types.
```suggestion
	if !same {
		// both are pointers but not the same type & pointing to the same address
```